src/main/java/com/redhat/microprofile/psi/internal/quarkus/core/properties/QuarkusConfigMappingProvider.java

```diff
@@ -145,19 +146,20 @@ private void populateConfigObject(PsiClass configMappingType, String prefixStr,
 				PsiType psiType = method.getReturnType();
 				String returnTypeSignature = getResolvedResultTypeName(method);
 				String resolvedTypeSignature = getRawResolvedTypeName(method);
-				if (isOptional(resolvedTypeSignature)) {
+				if (isGenericOptional(psiType)) {
```

Comment:
isGenericOptional call getTypeParameter and getTypeParameter is called again in the if.

I will keep the isOptional method (remove isGenericOptional method, and after that i will use  getTypeParameter

``` java
if (isOptional(resolvedTypeSignature)) {
 // Here we are in the case:
  // - java.util.Optional<String>
  // - java.util.OptionalInt
  PsiType psiTypeParameter = getTypeParameter(psiType);
  if (psiTypeParameter  != null) {
    // java.util.Optional<String>
    psiType = psiTypeParameter ; 
   resolvedTypeSignature = getRawResolvedTypeName(psiType);
   }

```